Formula/llvm@7.rb

```diff
@@ -0,0 +1,224 @@
+
+  resource "libcxx" do
+    url "https://releases.llvm.org/7.0.1/libcxx-7.0.1.src.tar.xz"
+    sha256 "020002618b319dc2a8ba1f2cba88b8cc6a209005ed8ad29f9de0c562c6ebb9f1"
+  end
+
+  resource "libunwind" do
+    url "https://releases.llvm.org/7.0.1/libunwind-7.0.1.src.tar.xz"
+    sha256 "89c852991dfd9279dbca9d5ac10b53c67ad7d0f54bbab7156e9f057a978b5912"
+  end
+
+  resource "lld" do
+    url "https://releases.llvm.org/7.0.1/lld-7.0.1.src.tar.xz"
+    sha256 "8869aab2dd2d8e00d69943352d3166d159d7eae2615f66a684f4a0999fc74031"
+  end
+
+  resource "openmp" do
+    url "https://releases.llvm.org/7.0.1/openmp-7.0.1.src.tar.xz"
+    sha256 "bf16b78a678da67d68405214ec7ee59d86a15f599855806192a75dcfca9b0d0c"
+  end
+
+  resource "polly" do
+    url "https://releases.llvm.org/7.0.1/polly-7.0.1.src.tar.xz"
+    sha256 "1bf146842a09336b9c88d2d76c2d117484e5fad78786821718653d1a9d57fb71"
+  end
+
+  def install
+    # Apple's libstdc++ is too old to build LLVM
+    ENV.libcxx if ENV.compiler == :clang
+
+    (buildpath/"tools/clang").install resource("clang")
+    (buildpath/"tools/clang/tools/extra").install resource("clang-extra-tools")
+    (buildpath/"projects/openmp").install resource("openmp")
+    (buildpath/"projects/libcxx").install resource("libcxx")
+    (buildpath/"projects/libunwind").install resource("libunwind")
+    (buildpath/"tools/lld").install resource("lld")
+    (buildpath/"tools/polly").install resource("polly")
+    (buildpath/"projects/compiler-rt").install resource("compiler-rt")
+
+    # compiler-rt has some iOS simulator features that require i386 symbols
+    # I'm assuming the rest of clang needs support too for 32-bit compilation
+    # to work correctly, but if not, perhaps universal binaries could be
+    # limited to compiler-rt. llvm makes this somewhat easier because compiler-rt
+    # can almost be treated as an entirely different build from llvm.
+    ENV.permit_arch_flags
+
+    args = %W[
+      -DLIBOMP_ARCH=x86_64
+      -DLINK_POLLY_INTO_TOOLS=ON
+      -DLLVM_BUILD_EXTERNAL_COMPILER_RT=ON
+      -DLLVM_BUILD_LLVM_DYLIB=ON
+      -DLLVM_ENABLE_EH=ON
+      -DLLVM_ENABLE_FFI=ON
+      -DLLVM_ENABLE_LIBCXX=ON
+      -DLLVM_ENABLE_RTTI=ON
+      -DLLVM_INCLUDE_DOCS=OFF
+      -DLLVM_INSTALL_UTILS=ON
+      -DLLVM_OPTIMIZED_TABLEGEN=ON
+      -DLLVM_TARGETS_TO_BUILD=all
+      -DWITH_POLLY=ON
+      -DFFI_INCLUDE_DIR=#{Formula["libffi"].opt_lib}/libffi-#{Formula["libffi"].version}/include
+      -DFFI_LIBRARY_DIR=#{Formula["libffi"].opt_lib}
+      -DLLVM_CREATE_XCODE_TOOLCHAIN=ON
+    ]
+
+    mkdir "build" do
+      system "cmake", "-G", "Unix Makefiles", "..", *(std_cmake_args + args)
+      system "make"
+      system "make", "install"
+      system "make", "install-xcode-toolchain"
+    end
+
+    (share/"clang/tools").install Dir["tools/clang/tools/scan-{build,view}"]
+    (share/"cmake").install "cmake/modules"
+    inreplace "#{share}/clang/tools/scan-build/bin/scan-build", "$RealBin/bin/clang", "#{bin}/clang"
```

Comment:
It seems that we need to apply #37028 for versioned LLVM formula.

Reply:
Thanks for the information. I've fixed it.